### PR TITLE
Copy flutter.bat before execution

### LINKED ISSUE #14578: Error when flutter upgrade or flutter channel modifies flutter.bat
On my Windows RS2 (v10.0.14393) machine, previously running the alpha channel, I attempted to switch to the beta channel. Error message as below...

```
[C:\Users\timsneath] flutter doctor
[√] Flutter (on Microsoft Windows [Version 10.0.14393], locale en-US, channel alpha)
    • Flutter version 0.0.21 at c:\git\flutter
    • Framework revision 2e449f06f0 (10 days ago), 2018-01-29 14:26:51 -0800
    • Engine revision 6921873c71
    • Tools Dart version 2.0.0-dev.16.0
    • Engine Dart version 2.0.0-edge.da1f52592ef73fe3afa485385cb995b9aec0181a

[√] Android toolchain - develop for Android devices (Android SDK 27.0.2)
    • Android SDK at C:\Users\timsneath\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-27, build-tools 27.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[√] Android Studio (version 3.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[√] Connected devices
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 8.1.0 (API 27) (emulator)


[C:\Users\timsneath] flutter channel
Flutter channels:
* alpha
  beta
  dev
  master

[C:\Users\timsneath] flutter channel beta
Switching to flutter channel 'beta'...
git: Branch 'beta' set up to track remote branch 'beta' from 'origin'.
git: Switched to a new branch 'beta'
'ot_pathflutter_tools_dirscript_path"' is not recognized as an internal or external command,
operable program or batch file.
Downloading package sky_engine...                     0.5s
Downloading common tools...                           0.4s
Downloading windows-x64 tools...                      1.0s
Downloading windows-x64 tools...                      1.3s
Downloading android-arm-profile/windows-x64 tools... 3650.2s
Downloading android-arm-release/windows-x64 tools...  0.4s
Downloading android-x86 tools...                      0.7s
Downloading android-x64 tools...                      0.8s
Downloading android-arm tools...                      0.6s
Downloading android-arm-profile tools...              0.5s
Downloading android-arm-release tools...              0.5s
Switching to flutter channel 'beta'...
git: M  VERSION
git: Your branch is up to date with 'origin/beta'.
git: Already on 'beta'

[C:\Users\timsneath]
```

### PR DESCRIPTION
**Pull request by goderbauer**
Fixes https://github.com/flutter/flutter/issues/14578

### CODE REVIEW

**Review by aam — APPROVED** [review]

**Comment by Hixie** [comment]
Can we put it in `bin/cache/`?

File: bin/flutter.bat [hunk 1]
```diff
@@ -12,6 +12,15 @@ REM work across all platforms!
 REM
 REM --------------------------------------------------------------------------
 
+REM Execute a copy of the script to prevent modifications to a running script,
+REM see https://github.com/flutter/flutter/issues/14578 for details.
```
Comment (Hixie) [inline bin/flutter.bat]: I don't think you need the reference to the bug, just say something like "(as might happen during an upgrade)" or some such.

**Comment by goderbauer** [comment]
> Can we put it in bin/cache/?

Initially I wanted to avoid that because it adds more complexity to this delicate script (what if cache doesn't exist yet? Now the original script and the copied version have different parent directories. etc.)

However, thinking about this more I am going to re-work this and break the script up into two scripts: one flutter.bat which is a wrapper that just copies a to-be-created script from bin/internal to bin/cache and then executes it. The new script will then do the heavy lifting of initializing flutter.

**Comment by goderbauer** [comment]
I made the changes as described above. PTAL

**Review by aam — APPROVED** [review]
Still lgtm, even though it's a little sad to see similarity between bin/fluttter and bin/flutter.bat lost(with bin/flutter.bat split into two).

**Pull request merged**